index.js

```diff
@@ -545,15 +545,10 @@ class Analytics {
   flush(callback) {
     // check if earlier flush was pushed to queue
     this.logger.debug("in flush");
-    if (this.state == "running") {
-      this.logger.debug("skipping flush, earlier flush in progress");
-      return;
-    }
-    this.state = "running";
```

Comment:
Let's not remove the states completely.

Reply:
what is the benefit of keeping it? I mean in the future it will create more confusion.

Reply:
Agreed. It's a carefully thought-out state machine so didn't want to wipe it off. Add comments at its definition that it is not used.

Reply:
Done